### PR TITLE
Revert ykpack dep and fix opts

### PR DESCRIPTION
**Pull request by vext01**
This completes the 3-phase merge currently in-situ.

I also sneaked the fix for tracing and optimisations in at the same time. Hope that is ok.

### CODE REVIEW

File: src/librustc/session/config.rs [hunk 1]
```diff
@@ -2344,10 +2344,10 @@ fn parse_opt_level(
         }
     };
 
-    // Disable hardware tracing when optimisations are enabled.
-    // https://github.com/softdevteam/ykrustc/issues/66
-    if opt_level != OptLevel::No && cg.tracer == TracerMode::Hardware {
-       cg.tracer = TracerMode::Software;
+    // Currently we disallow optimisation *and* tracing, as we don't want LLVM to re-order things
```
Comment (ltratt) [inline src/librustc/session/config.rs]: This doesn't read well: I think we want to say something like "We don't support both optimisation and tracing being enabled simultaneously as LLVM is likely to reorder things, destroying the correctness of our labels".

Comment (vext01) [reply src/librustc/session/config.rs]: How's my last push?

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat

**Comment by vext01** [comment]
I'm going to skip those failing swt tests. Just checking if any others fail...

**Comment by vext01** [comment]
That last push fixes the tests.

**Comment by ltratt** [comment]
Please squash.

**Comment by vext01** [comment]
splat.

**Pull request merged**